app/build.gradle

```diff
@@ -229,16 +232,40 @@ android {
   }
 
   afterEvaluate {
-    def buildTypes = [
+    def assembleReleaseTasks = [
         "assembleStagingRelease",
         "assembleSandboxRelease",
         "assembleProductionRelease",
         "assembleSecurityRelease"
     ]
 
-    buildTypes.forEach { buildType ->
+    assembleReleaseTasks.forEach { buildType ->
       tasks.findByName(buildType).finalizedBy(deleteProguardMappings)
     }
+
+    def kaptTasks = [
+        'kaptSandboxReleaseKotlin'   : 'sandboxRelease',
+        'kaptStagingReleaseKotlin'   : 'stagingRelease',
+        'kaptSecurityReleaseKotlin'  : 'securityRelease',
+        'kaptProductionReleaseKotlin': 'productionRelease'
+    ]
+
+    kaptTasks.each { buildType, sourceSetName ->
+      def taskQualifier = buildType
+          .replaceAll('kapt', '')
+          .replaceAll('Kotlin', '')
+
+      def taskName = "generate${taskQualifier}RoomMetadata"
+
+      def generateTask = task "$taskName"(type: GenerateRoomMetadataTask) {
+        sourceSet = sourceSetName
+        csvAssetName = 'db_metadata.csv'
+      }
+      tasks.findByName(buildType).finalizedBy(generateTask)
+      tasks.named("merge${taskQualifier}Assets").configure {
```

Comment:
@msasikanth This was the change I made. There were cases where Gradle would execute this task _before_ the metadata generation task ran so it would not get included in the build. This ensures that it runs after.